pkg/endpoint/regeneration/wait.go

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package regeneration
+
+import (
+	"context"
+
+	"github.com/cilium/hive/cell"
+
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+// InitWaitGroup for delaying the initial endpoint regeneration.
+type InitWaitGroup interface {
```

Comment:
Suggestion: expose `lock.InitWaitGroup` interface, then individual timing points just wrap the interface.